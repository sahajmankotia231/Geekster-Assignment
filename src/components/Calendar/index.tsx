import React, { ReactElement } from "react";
import styled from "styled-components";

import { ReactComponent as IconCalendar } from "assets/calendar.svg";
import { BORDER_GRAY } from "constants/colors";
import { INPUT_RADIUS } from "constants/styles";

const CalendarContainer = styled.div`
  width: 100%;
  height: 40px;
  align-items: center;
  border: 1px solid ${BORDER_GRAY};
  border-radius: ${INPUT_RADIUS}px;
  display: flex;
  font-size: 15px;
  justify-content: space-between;
  line-height: 24px;
  padding: 12px 24px 12px 12px;
`;

const Pipe = styled.span`
  height: 24px;
  width: 1px;
  background-color: ${BORDER_GRAY};
  display: block;
`;

const Calendar = (): ReactElement => (
  <CalendarContainer>
    <IconCalendar />
    <strong>Check-in</strong>
    <Pipe />
    <strong>Check-out</strong>
  </CalendarContainer>
);

export default Calendar;
