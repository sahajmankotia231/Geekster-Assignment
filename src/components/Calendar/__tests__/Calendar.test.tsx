import React from "react";
import { render } from "@testing-library/react";

import Calendar from "..";

describe("<Calendar />", () => {
  it("should render properly", () => {
    const { container } = render(<Calendar />);
    expect(container.firstChild).toMatchSnapshot();
  });
});
