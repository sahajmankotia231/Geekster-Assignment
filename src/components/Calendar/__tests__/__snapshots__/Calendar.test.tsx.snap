// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Calendar /> should render properly 1`] = `
<div
  class="sc-bdvvaa gBaEPd"
>
  <svg>
    calendar.svg
  </svg>
  <strong>
    Check-in
  </strong>
  <span
    class="sc-gsDJrp cEMzuB"
  />
  <strong>
    Check-out
  </strong>
</div>
`;
